src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/create_index_pattern_wizard.js

```diff
@@ -0,0 +1,284 @@
+  };
+
+  this.documentationLinks = documentationLinks;
+
+  // Configure the new index pattern we're going to create.
+  this.formValues = {
+    name: '',
+    expandWildcard: false,
+    timeFieldOption: loadingOption,
+  };
+
+  // UI state.
+  this.timeFieldOptions = [
+    loadingOption,
+  ];
+  this.wizardStep = 'indexPattern';
+  this.isFetchingExistingIndices = true;
+  this.isFetchingMatchingIndices = false;
+  this.isFetchingTimeFieldOptions = false;
+  this.isCreatingIndexPattern = false;
+  this.doesIncludeSystemIndices = false;
+  let allIndices = [];
+  let matchingIndices = [];
+  let partialMatchingIndices = [];
+  this.allIndices = [];
+  this.matchingIndices = [];
+  this.partialMatchingIndices = [];
+
+  function createReasonableWait() {
+    return new Promise(resolve => {
+      // Make every fetch take a set amount of time so the user gets some feedback that something
+      // is happening.
+      $timeout(() => {
+        resolve();
+      }, 500);
+    });
+  }
+
+  const whiteListIndices = indices => {
+    if (!indices) {
+      return indices;
+    }
+
+    if (this.doesIncludeSystemIndices) {
+      return indices;
+    }
+
+    // All system indices begin with a period.
+    return indices.filter(index => (
+      index.indexOf('.') !== 0
+    ));
+  };
+
+  const updateWhiteListedIndices = () => {
+    this.allIndices = whiteListIndices(allIndices);
+    this.matchingIndices = whiteListIndices(matchingIndices);
+    this.partialMatchingIndices = whiteListIndices(partialMatchingIndices);
+  };
+
+  this.onIncludeSystemIndicesChange = () => {
+    updateWhiteListedIndices();
+  };
+
+  let mostRecentFetchMatchingIndicesRequest;
+
+  this.fetchMatchingIndices = () => {
+    this.isFetchingMatchingIndices = true;
+
+    // Default to searching for all indices.
+    const exactSearchQuery = this.formValues.name;
+    let partialSearchQuery = this.formValues.name;
+
+    if (!_.endsWith(partialSearchQuery, '*')) {
+      partialSearchQuery = `${partialSearchQuery}*`;
+    }
+    if (!_.startsWith(partialSearchQuery, '*')) {
+      partialSearchQuery = `*${partialSearchQuery}`;
+    }
+
+    const thisFetchMatchingIndicesRequest = mostRecentFetchMatchingIndicesRequest = Promise.all([
+      indicesService.getIndices(exactSearchQuery),
+      indicesService.getIndices(partialSearchQuery),
+      createReasonableWait(),
+    ])
+    .then(([
+      matchingIndicesResponse,
+      partialMatchingIndicesResponse,
+    ]) => {
+      if (thisFetchMatchingIndicesRequest === mostRecentFetchMatchingIndicesRequest) {
+        matchingIndices = matchingIndicesResponse.sort();
+        partialMatchingIndices = partialMatchingIndicesResponse.sort();
+        updateWhiteListedIndices();
+        this.isFetchingMatchingIndices = false;
+      }
+    });
+  };
+
+  this.fetchExistingIndices = () => {
+    this.isFetchingExistingIndices = true;
+    Promise.all([
+      indicesService.getIndices('*'),
```

Comment:
This method doesn't return indices accessible via cross-cluster search, and I'm not certain we will be able to. Since the only APIs that are cross-cluster search enabled are the `_search` and `_field_caps` APIs, we might be able to use the `_search` API and do a terms aggregation on `_index`, but that might have prohibitive performance characteristics and is blocked on https://github.com/elastic/elasticsearch/issues/25606 anyway.